src/gurt/misc.c

```diff
@@ -513,14 +518,15 @@ d_sgl_fini(d_sg_list_t *sgl, bool free_iovs)
 {
 	int	i;
 
-	if (sgl->sg_iovs == NULL)
+	if (sgl == NULL || sgl->sg_iovs == NULL)
 		return;
 
 	for (i = 0; free_iovs && i < sgl->sg_nr; i++)
 		D_FREE(sgl->sg_iovs[i].iov_buf);
 
 	D_FREE(sgl->sg_iovs);
-	memset(sgl, 0, sizeof(*sgl));
+	sgl->sg_nr_out = 0;
```

Comment:
Can you change handling of free_iovs on L524, if free_iovs is false then it shouldn't do the for loop.

Reply:
It don't do the loop if free_iovs is false. Do you mean just make the code more readable like following?

```
if (free_iovs)
	for (i = 0; i < sgl->sg_nr; i++)
		D_FREE(sgl->sg_iovs[i].iov_buf);
```

Reply:
Yes.  The compiler will probably optimise it to be the same anyway but it's a lot more readable like that.

Reply:
ok